src/Game.java

```diff
@@ -1,45 +1,119 @@
 
 public class Game {
-    public static void main(String[] args) {
-        System.out.println("Welcome to PPTHangman!");
-        System.out.println("Please enter the number of players:");
 
-        Scanner scanner = new Scanner(System.in);
+  public static Map<String, Integer> leaderboardToSet(List<Player> allPlayers) {
+    return allPlayers.stream()
+        .collect(Collectors.toMap(Player::toString, Player::getScore));
+  }
 
-        int numPlayers = Integer.parseInt(scanner.nextLine());
-        List<Player> allPlayers = new ArrayList<>();
+  public static void main(String[] args) {
+    System.out.println("Welcome to PPTHangman!");
+    System.out.println("Please enter the number of players:");
 
-        for (int i = 0; i < numPlayers; i++) {
-            System.out.println("Please enter the name of the " + (i + 1) + "th player");
-            allPlayers.add(new Player(scanner.nextLine()));
+    Scanner scanner;
+    int numPlayers;
+    String numPlayersString;
+
+    while (true) {
+      scanner = new Scanner(System.in);
+      numPlayersString = scanner.nextLine();
+
+      if (numPlayersString.length() == 0){
+        System.out.println("Invalid input, please insert a number:");
+      } else {
+        numPlayers = Integer.parseInt(numPlayersString);
+        if (numPlayers > 1) {
+          break;
+        } else if (numPlayers == 1) {
+          System.out.println(
+              "You cannot play alone! Please enter a number greater than 1:");
+        } else {
+          System.out.println("Please enter a number greater than 1:");
+        }
+      }
+    }
+
+    List<Player> allPlayers = new ArrayList<>();
+
+    for (int i = 0; i < numPlayers; i++) {
+      System.out.println("Please enter the name of player " + (i + 1) + ":");
+      String playerName;
+      while (true) {
+        playerName = scanner.nextLine();
+        if (!allPlayers.stream().map(Object::toString)
+            .collect(Collectors.toList()).contains(playerName)) {
+          if (playerName.length() != 0) {
+            break;
+          }
+          System.out.println("Invalid name, please enter another name");
+        } else {
+          System.out.println("Please enter another name, " + playerName
+              + " has already been taken:");
         }
+      }
 
-        int currentLeaderIndex = 0;
+      allPlayers.add(new Player(playerName));
+    }
+
+    int currentLeaderIndex = 0;
+
+    mainLoop:
+    while (true) {
 
-        while (true) {
-            System.out.println("Do you wish to start a new game? Y/N");
-            String answer = scanner.nextLine();
+      Player leader = allPlayers.get(currentLeaderIndex);
 
-            if (answer.startsWith("N")) {
-                break;
-            }
+      Word word;
+      System.out.println(leader + ", you are the leader, please pick a word:");
 
-            Player leader = allPlayers.get(currentLeaderIndex);
+      while (true) {
+        word = new Word(scanner.nextLine());
```

Comment:
If you took the advice I gave in the above comment, you could replace this whole while loop with one call:

`word = new Word(askUntilCondition("Please enter a word with at least 1 letter:", input -> input.length() != 0 && input.length() < 30 && ...)` etc.